test/index.test.js

```diff
@@ -1091,44 +1091,79 @@ describe('Segment.io', function() {
           describe('deleteCrossDomainId', function() {
             it('should not delete cross domain identifiers by default', function() {
               segment.cookie('seg_xid', 'test_xid');
+              segment.cookie('seg_xid_ts', 'test_xid_ts');
+              segment.cookie('seg_xid_fd', 'test_xid_fd');
               analytics.identify({
                 crossDomainId: 'test_xid'
               });
 
               segment.deleteCrossDomainIdIfNeeded();
 
               assert.equal(segment.cookie('seg_xid'), 'test_xid');
+              assert.equal(segment.cookie('seg_xid_ts'), 'test_xid_ts');
+              assert.equal(segment.cookie('seg_xid_fd'), 'test_xid_fd');
               assert.equal(analytics.user().traits().crossDomainId, 'test_xid');
             });
 
             it('should do not delete cross domain identifiers if disabled', function() {
               segment.options.deleteCrossDomainId = false;
 
               segment.cookie('seg_xid', 'test_xid');
+              segment.cookie('seg_xid_ts', 'test_xid_ts');
+              segment.cookie('seg_xid_fd', 'test_xid_fd');
               analytics.identify({
                 crossDomainId: 'test_xid'
               });
 
               segment.deleteCrossDomainIdIfNeeded();
 
               assert.equal(segment.cookie('seg_xid'), 'test_xid');
+              assert.equal(segment.cookie('seg_xid_ts'), 'test_xid_ts');
+              assert.equal(segment.cookie('seg_xid_fd'), 'test_xid_fd');
               assert.equal(analytics.user().traits().crossDomainId, 'test_xid');
             });
 
             it('should delete cross domain identifiers if enabled', function() {
               segment.options.deleteCrossDomainId = true;
 
               segment.cookie('seg_xid', 'test_xid');
+              segment.cookie('seg_xid_ts', 'test_xid_ts');
+              segment.cookie('seg_xid_fd', 'test_xid_fd');
               analytics.identify({
                 crossDomainId: 'test_xid'
               });
 
               segment.deleteCrossDomainIdIfNeeded();
 
               assert.equal(segment.cookie('seg_xid'), null);
+              assert.equal(segment.cookie('seg_xid_ts'), null);
+              assert.equal(segment.cookie('seg_xid_fd'), null);
               assert.equal(analytics.user().traits().crossDomainId, null);
             });
 
+            it('should delete xid cookie if only cookie exists', function() {
+              segment.options.deleteCrossDomainId = true;
+
+              analytics.identify({
+                crossDomainId: 'test_xid'
+              });
+
+              segment.deleteCrossDomainIdIfNeeded();
+
+              assert.equal(analytics.user().traits().crossDomainId, null);
+            });
+
+            it('should delete localStorage trait if only traits exists', function() {
```

Comment:
this test case doesn't test what it's supposed to test.

Reply:
sorry - fixed now!